code/Weapons/Gadget/Components/AirstrikeGadgetComponent.cs

```diff
@@ -0,0 +1,111 @@
+using Grubs.Utils;
+
+namespace Grubs;
+
+[Prefab]
+public partial class AirstrikeGadgetComponent : GadgetComponent
+{
+	[Prefab]
+	public Prefab Projectile { get; set; }
+
+	[Prefab]
+	public float DropRateSeconds { get; set; } = 0.2f;
+
+	[Prefab]
+	public int ProjectileCount { get; set; } = 1;
+
+	/// <summary>
+	/// Usually only Vector3.Backward or Vector3.Forward.
+	/// Vector3.Forward for Left -> Right and vice-versa.
+	/// </summary>
+	[Net]
+	public Vector3 BombingDirection { get; set; }
+
+	// Network this so the client can play the bombing animation at the right time.
+	[Net]
+	public Vector3 TargetPosition { get; set; }
+
+	public bool HasReachedTarget { get; private set; }
+
+	public const float SpawnOffsetX = 3000;
+	public const float SpawnOffsetY = -30;
+	public const float SpawnOffsetZ = 64;
+
+	private Particles _trail;
+	private Sound _engineSound;
+	private const float _dropPayloadDistanceThreshold = 950;
+	private const float _planeFlySpeed = 23;
+
+	public override void ClientSpawn()
+	{
+		_trail = Particles.Create( "particles/contrails/contrail_base.vpcf", Gadget, "trail" );
+		_engineSound = Entity.SoundFromScreen( "sounds/airstrike/plane_engine_loop.sound" );
+	}
+
+	public override void Simulate( IClient client )
+	{
+		Gadget.Position += BombingDirection * _planeFlySpeed;
+		var distanceToTarget = Gadget.Position.WithY( 0 ).WithZ( 0 ).Distance( TargetPosition.WithY( 0 ).WithZ( 0 ) );
+
+		if ( distanceToTarget <= _dropPayloadDistanceThreshold + 320 )
+			Animate();
+
+		if ( distanceToTarget <= _dropPayloadDistanceThreshold && !HasReachedTarget )
+		{
+			HasReachedTarget = true;
+
+			if ( Game.IsServer )
+			{
+				Gadget.ShouldCameraFollow = false;
+				DropPayload();
+			}
+
+			if ( Game.IsClient )
+			{
+				_engineSound.FadeOut( 0.150f, fadeTime: 1.4f, fadeMultiplier: 2.5f );
+				Entity.SoundFromScreen( "sounds/airstrike/plane_bay_door_open.sound" );
+			}
+		}
+
+		if ( HasReachedTarget && Gadget.Position.WithY( 0 ).WithZ( 0 ).Distance( GrubsGame.Instance.Terrain.Position.WithY( 0 ).WithZ( 0 ) ) >= GrubsConfig.TerrainLength * 3 )
```

Comment:
Maybe we can wrap up `Gadget.Position.WithY( 0 ).WithZ( 0 ).Distance( GrubsGame.Instance.Terrain.Position.WithY( 0 ).WithZ( 0 ) ) >= GrubsConfig.TerrainLength * 3` to a variable that explain what it means?